apps/prairielearn/src/sprocs/sync_assessments.sql

```diff
@@ -450,45 +450,25 @@ BEGIN
-                        FOR valid_group_role IN (
-                            SELECT gr.id, gr.role_name
-                            FROM group_roles as gr
-                            WHERE gr.assessment_id = new_assessment_id
-                        ) LOOP
-                            -- Insert roles that can view
-                            INSERT INTO assessment_question_role_permissions (
-                                assessment_question_id,
-                                group_role_id,
-                                can_view
-                            ) VALUES (
-                                new_assessment_question_id,
-                                valid_group_role.id,
-                                (valid_group_role.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_view')))
-                            ) ON CONFLICT (assessment_question_id, group_role_id)
-                            DO UPDATE
-                            SET
-                                assessment_question_id = EXCLUDED.assessment_question_id,
-                                group_role_id = EXCLUDED.group_role_id,
-                                can_view = EXCLUDED.can_view;
-
-                            -- Insert roles that can submit
-                            INSERT INTO assessment_question_role_permissions (
-                                assessment_question_id,
-                                group_role_id,
-                                can_submit
-                            ) VALUES (
-                                new_assessment_question_id,
-                                valid_group_role.id,
-                                (valid_group_role.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_submit')))
-                            ) ON CONFLICT (assessment_question_id, group_role_id)
-                            DO UPDATE
-                            SET
-                                assessment_question_id = EXCLUDED.assessment_question_id,
-                                group_role_id = EXCLUDED.group_role_id,
-                                can_submit = EXCLUDED.can_submit;
-                        END LOOP;
+                        INSERT INTO assessment_question_role_permissions (
+                            assessment_question_id,
+                            group_role_id,
+                            can_view,
+                            can_submit
+                        ) SELECT
+                            new_assessment_question_id AS assessment_question_id,
+                            gr.id AS group_role_id,
+                            (gr.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_view'))) AS can_view,
+                            (gr.role_name IN (SELECT * FROM JSONB_ARRAY_ELEMENTS_TEXT(assessment_question->'can_submit'))) AS can_submit
```

Comment:
The `AS` clauses are again unnecessary.

To be clear, my current thinking is that these unnecessary `AS` clauses are not just unnecessary but are in fact actively harmful. My thinking is that, assuming Postgres does actually just use column order to match between the `SELECT` and `INSERT`, then we always need to carefully check the order to make sure it's consistent. Having `AS` clauses might actually mislead us by making it look like things are ok when in fact they are not.

Thinking about this further, however, I can see an argument that we should first check each line in the `SELECT` to make sure it makes sense with its own `AS` clause, and then we should check that the `INSERT` columns match the `AS` names in order. Is this how you are thinking about it? I could buy that.

Reply:
You're exactly correct, I don't know where I got the idea that column names mattered for insertions. Real-life LLM hallucination moment? 😂

I think the risk of (incorrectly) assuming that column names matter means I should remove the names entirely. Thanks for pointing this out!

Reply:
Removed in e1e4645.